lighthouse-treemap/app/src/main.js

```diff
@@ -33,18 +33,15 @@ class TreemapViewer {
    * @param {HTMLElement} el
    */
   constructor(options, el) {
-    const treemapDebugData = /** @type {LH.Audit.Details.DebugData} */ (
+    const scriptTreemapData = /** @type {LH.Audit.Details.TreemapData} */ (
       options.lhr.audits['script-treemap-data'].details);
-    if (!treemapDebugData || !treemapDebugData.treemapData) {
+    if (!scriptTreemapData || scriptTreemapData.type !== 'treemap-data') {
```

Comment:
good call, and this check means the `LH.Audit.Details.TreemapData` cast above can be dropped too

Reply:
That caused some weird error... Can you try it and see?

Reply:
> That caused some weird error

ah, perfect, because finally I can see it :)

Looks like this was from my idea in https://github.com/GoogleChrome/lighthouse/pull/12483#discussion_r633659372

In theory `LH.RawIcu<LH.FormattedIcu<Audit.Details>>` should resolve to an identity transformation and leave `Audit.Details` alone. However when I suggested we add `dom?: HTMLElement` to `LH.Audit.Details.TreemapData`, that ends up unleashing the zalgo and really putting the test to `LH.RawIcu<LH.FormattedIcu<>>` actually being an identity transformation. If you add `"noErrorTruncation": true` to our tsconfig you end up with a very fun error message. I would post it here under `<details>` but it's 1.7MiB long, repeated once for good measure.

A selection from our `Audit.Details` monstrosity:
```js
...
spellcheck: boolean;
title: string | IcuMessage;
translate: boolean;
click: IcuMessage | {};
readonly classList: { [x: number]: string | IcuMessage; readonly length: number; value: string | IcuMessage; toString: IcuMessage | {}; add: IcuMessage | {}; contains: IcuMessage | {}; item: IcuMessage | {}; remove: IcuMessage | {}; replace: IcuMessage | {}; supports: IcuMessage | {}; toggle: IcuMessage | {}; forEach: IcuMessage | {}; entries: IcuMessage | {}; keys: IcuMessage | {}; values: IcuMessage | {}; };
className: string | IcuMessage;
readonly clientHeight: number
...
```

I think the best solution is just to keep `dom` out of the audit details type (since it can never be in the LHR anyways) and have `TreemapUtil.walk()` return `LH.Treemap.Node & {dom?: HTMLElement}` so that's still typed.

Reply:
wonder why we both had trouble seeing this locally initially... maybe the type calculation was too massive